templates/common-init-files/secret.yaml

```diff
@@ -25,6 +25,7 @@ stringData:
     set -euo pipefail
     export DEBIAN_FRONTEND=noninteractive
     hostnamectl set-hostname "$1" && hostname -F /etc/hostname
+    echo "$(ip a s eth1 |grep 'inet ' |cut -d' ' -f6|cut -d/ -f1) $1" >> /etc/hosts
```

Comment:
if the hostname of a node is an IP rather than the `label` as we are setting it now does that solve anything for us here?

Reply:
This is needed just in case of kubeadm to tell kubelet to use eth1's ip as node ip. Hostname is still the same. This makes `hostname -i` to return one ip address. If we don't set it, kubelet starts using the ip on eth0 as node ip.
```
oot@rah8-control-plane-24ggw:~# hostname
rah8-control-plane-24ggw
root@rah8-control-plane-24ggw:~# hostname -i
10.0.0.2
root@rah8-control-plane-24ggw:~# 
```

Reply:
could we use the `--node-ip` flag instead of this to set the IP of the node?

Reply:
`kubeadm init` doesn't have the `--node-ip` flag. It can be set for kubelet once kubelet is running but the certs get generated by kubeadm using the default ip it detects when its started the first time. I don't see any option to provide node-ip using a kubelet config file as its not supported using config file https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/. Not sure if there is a way we can specify the node-ip with `--cloud-provider=external` during kubeadm init phase to get things working.

Reply:
It looks like this will work but we need to get the node-ip here somehow:
```
apiVersion: kubeadm.k8s.io/v1beta3
kind: InitConfiguration
nodeRegistration:
  kubeletExtraArgs:
    node-ip: 10.0.0.2
```

Reply:
Updated the PR. PTAL 🙏